src/jaegertracing/samplers/SamplerTest.cpp

```diff
@@ -39,7 +39,7 @@ constexpr auto kTestOperationName = "op";
 constexpr auto kTestFirstTimeOperationName = "firstTimeOp";
 constexpr auto kTestDefaultSamplingProbability = 0.5;
 constexpr auto kTestMaxID =
-    static_cast<long double>(std::numeric_limits<uint64_t>::max()) / 2.0 + 1;
+    std::numeric_limits<uint64_t>::max() / 2 + 1;
```

Comment:
@rnburn can explain the issue here (#6).

Reply:
Unless I am missing something, in the issue #6, the problem occurs because we cast to double, which rounds up to a higher value. This latter, when casted back to  uint64_t cause the overflow.
Here, we are dividing an unsigned int by 2. I dont see why it can (over/under)flow.

Reply:
Indeed you are correct. My mistake.